### PR TITLE
Added configuration options for max memory per vm

### PR DESCRIPTION
**Pull request by jonasfj**
This added an engine option to QEMU engine so we can configure max allowed memory per VM.

If  a machine is specified to require more memory than is allowed, it'll result in an malformed-payload error.

### CODE REVIEW

File: engines/qemu/vm/machine.go [hunk 1]
```diff
@@ -41,6 +42,10 @@ const machineSchemaString = `{
 			"pattern":								"^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$",
 			"description":						"System UUID for the virtual machine"
 		},
+		"memory": {
+			"type":										"integer",
```
Comment (walac) [inline engines/qemu/vm/machine.go]: is this some indent issue or a github thing?

File: engines/qemu/vm/options.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                  these limitations will be resolved 'malformed-payload'.`,
+	},
+	Properties: schematypes.Properties{
+		"maxMemory": schematypes.Integer{
+			MetaData: schematypes.MetaData{
+				Title: "Max Memory",
+				Description: `Maximum allowed virtual machine memory in MiB. This is
+										 	also the default memory if the machine image doesn't
+											specify memory requirements. If the machine specifies a
+											memory higher than this, the task will fail to run.`,
```
Comment (walac) [inline engines/qemu/vm/options.go]: nit: indent problems here

File: engines/qemu/vm/vm.go [hunk 3]
```diff
@@ -42,11 +43,25 @@ type VirtualMachine struct {
 	domain       *qemu.Domain
 }
 
-// NewVirtualMachine constructs a new virtual machine.
+// NewVirtualMachine constructs a new virtual machine using the given
+// machineOptions, image, network and cdroms.
+//
+// Returns engines.MalformedPayloadError, if machineOptions and image definition
```
Comment (walac) [inline engines/qemu/vm/vm.go]: nit s/,//

File: engines/qemu/vm/machine.go (hunk 1 above)
Comment (jonasfj) [inline engines/qemu/vm/machine.go]: oh, my go editor uses tabs in strings and this happens... :)

**Pull request merged**